server/routes/routes.js

```diff
@@ -13,31 +13,44 @@ const router = (app) => {
   // Load tasks (users.auth_id).
   app.get("/api/tasks", (req, res) => {
     // console.log(req.session.passport.user);
-    pool
-      .query("SELECT * FROM task WHERE user_id = $1", [2])
-      .then((result) => {
-        res.status(200).json(result.rows);
-      })
-      .catch((e) => console.error(e));
+    const auth_id = req.session.passport.user; 
+  pool
+    .query("SELECT * FROM users WHERE auth_id = $1", [auth_id])
+    .then((result) => {
+      const user_id = result.rows[0].id;
+      // console.log(user_id)
+      pool
+        .query("SELECT * FROM task WHERE user_id = $1", [user_id])
+        .then((result) => {
+          // console.log(result.rows);
+          res.status(200).json(result.rows);
+        })
+        .catch((e) => console.error(e));
+    })
+    .catch((e) => console.error(e));
   });
 
   // Add user tasks (users.auth_id).
   app.post("/api/tasks", (req, res) => {
     // console.log(req.session.passport.user);
-
-    const user_id = 2;
-    const task_subject = req.body.task_subject;
-    const subject_description = req.body.subject_description;
-    const sub_task_option = req.body.sub_task_option;
-    const sub_tasks = req.body.sub_tasks;
-    const reward = req.body.reward;
-    const resources = req.body.resources;
-    const by_time = req.body.by_time;
-    const by_day = req.body.by_day;
-
+const auth_id = req.session.passport.user;
+const task_subject = req.body.task_subject;
+const subject_description = req.body.subject_description;
+const reward = req.body.reward;
+const resources = req.body.resources;
+const by_time = req.body.by_time;
+const by_date = req.body.by_date;
+const sub_task_option = req.body.sub_task_option;
+const sub_tasks = req.body.sub_tasks;
```

Comment:
It may be possible to condense lines 37 to 44 into a single line through destructuring.

Reply:
Thanks andy for your review. I think It would be a very long line and it may cause confusion to the reader.

Reply:
Ok fair enough, it could have been split into two lines :)
```
const [ task_subject, subject_description, reward, resources ] = req.body;
const [ by_time, by_date, sub_task_option, sub_task ] = req.body;
```

Reply:
ok got you know I will do it